File: Backend/StockDaddy.Application/DTOs/UserDto.cs
namespace StockDaddy.Application.DTOs;

public class UserDto
{
    public Guid Id { get; set; }
    public Guid TenantId { get; set; }
    public Guid RoleId { get; set; }
    public Guid? StoreId { get; set; }

    public string Username { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public bool IsDeleted { get; set; }
    public DateTime? DeletedAt { get; set; }
}

File: Frontend/src/dtos/user.dto.ts
export interface UserStoreAssignmentDto {
  storeId: number;
  roleId: number;
  isDefault: boolean;
  storeName?: string;
  roleName?: string;
}

export interface UserManagementDto {
  id: number;
  tenantId: number;
  roleId: number;
  storeId?: number;
  storeIds?: number[];
  defaultStoreId?: number;
  storeAssignments?: UserStoreAssignmentDto[];
  username: string;
  email: string;
  createdAt: string;
  updatedAt: string;
  isDeleted: boolean;
}

export interface UpdateUserManagementRequest {
  roleId: number;
  storeId?: number;
  storeIds?: number[];
  defaultStoreId?: number;
  storeAssignments?: UserStoreAssignmentDto[];
  username: string;
  email: string;
  passwordHash?: string;
}

export interface CreateUserManagementRequest {
  tenantId: number;
  roleId: number;
  storeId?: number;
  storeIds?: number[];
  defaultStoreId?: number;
  storeAssignments?: UserStoreAssignmentDto[];
  username: string;
  email: string;
  passwordHash: string;
}

export interface AssignUserStoreAssignmentsRequest {
  defaultRoleId?: number;
  defaultStoreId?: number;
  assignments: UserStoreAssignmentDto[];
}
